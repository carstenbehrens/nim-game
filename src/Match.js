import React from 'react';
import styled from 'styled-components';

const SingleMatchContainer = styled.div`
	height: 8rem;
	width: 4rem;
	cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #243d39;

  @media (min-width: 900px) {
    &:hover {
      background: #335751;
    }
	}
`;

const MatchBottom = styled.div`
	height: 5rem;
	width: 1rem;
	background: #915535;
  cursor: pointer;
  opacity: ${props => props.visible ? '100' : '0'};
`;

const MatchTip = styled.div`
  height: 1rem;
  width: 1rem;
  background: red;
  opacity: ${props => props.visible ? '100' : '0'};
`

export default ({ visible, matchNumber, onClick }) => {
	return (
		<SingleMatchContainer visible={visible} onClick={() => onClick(matchNumber)}>
      <MatchTip visible={visible} />
			<MatchBottom visible={visible} /> 
		</SingleMatchContainer>
	);
};
