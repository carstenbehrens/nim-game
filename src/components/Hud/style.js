import styled from 'styled-components'

export const HudContainer = styled.div`
	height: 30%;
	width: 100%;
	color: white;
	font-size: 2rem;
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: space-around;
`;

export const FinishTurn = styled.button`
	height: 4rem;
	width: 8rem;
	color: white;
	background: none;
	font-weight: bold;
	cursor: pointer;
	text-transform: uppercase;
`;