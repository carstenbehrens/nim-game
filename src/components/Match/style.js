import styled from "styled-components";

export const MatchContainer = styled.div`
  height: 8rem;
  width: 4rem;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: linear-gradient(
    180deg,
    rgba(0, 0, 0, 0) 0%,
    rgba(255, 255, 255, 0.1) 100%
  );
  border-radius: 3rem;
  -webkit-tap-highlight-color: transparent;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;

  @media (min-width: 900px) {
    &:hover {
      box-shadow: 0px 3px 46px -17px rgba(0, 0, 0, 0.3);
      background: linear-gradient(
        180deg,
        rgba(0, 0, 0, 0) 0%,
        rgba(255, 255, 255, 0.3) 100%
      );
    }
  }

  @media (max-width: 600px) {
    height: 90%;
  }
`;

export const MatchStick = styled.div`
  height: 5rem;
  width: 0.6rem;
  background: rgb(224, 175, 76);
  background: linear-gradient(
    270deg,
    rgba(224, 175, 76, 1) 0%,
    rgba(224, 175, 76, 1) 50%,
    rgba(230, 186, 102, 1) 50%,
    rgba(237, 201, 136, 1) 100%
  );
  cursor: pointer;
  opacity: ${props => (props.visible ? "100" : "0")};
`;

export const MatchTip = styled.div`
  height: 1.4rem;
  width: 1rem;
  border-radius: 40%;
  background: rgb(89, 21, 12);
  background: linear-gradient(
    126deg,
    rgba(89, 21, 12, 1) 0%,
    rgba(238, 141, 101, 1) 0%,
    rgba(193, 104, 66, 1) 37%,
    rgba(213, 49, 28, 1) 39%,
    rgba(71, 19, 12, 1) 100%
  );
  opacity: ${props => (props.visible ? "100" : "0")};
`;
