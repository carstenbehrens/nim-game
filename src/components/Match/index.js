import React from "react";
import { MatchContainer, MatchTip, MatchStick } from "./style";

export default ({ visible, number, onClick }) => {
  return (
    <MatchContainer onClick={() => onClick(number)}>
      <MatchTip visible={visible} />
      <MatchStick visible={visible} />
    </MatchContainer>
  );
};
