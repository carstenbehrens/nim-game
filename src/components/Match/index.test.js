import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Match from "./index";
import { MatchContainer, MatchStick } from "./style";

describe("Match", () => {
  test("renders without crashing", () => {
    shallow(<Match />);
  });

  test("matches snapshot", () => {
    const wrapper = mount(
      <Match visible={true} key={0} matchNumber={1} onClick={() => null} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("onClick gets called with the correct number", () => {
    MatchContainer.displayName = "MatchContainer";

    const handleClickMatch = jest.fn();

    const wrapper = shallow(<Match number={1} onClick={handleClickMatch} />);
    wrapper.find("MatchContainer").simulate("click");

    expect(handleClickMatch).toHaveBeenCalledWith(1);
  });

  test("visibility gets passed down", () => {
    MatchContainer.displayName = "MatchContainer";
    MatchStick.displayName = "MatchStick";

    const wrapper = shallow(<Match visible={false} />);

    expect(wrapper.find("MatchStick").props()).toEqual({ visible: false });
  });
});
