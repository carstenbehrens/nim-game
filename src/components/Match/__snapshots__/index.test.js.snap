// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match matches snapshot 1`] = `
<_default
  matchNumber={1}
  onClick={[Function]}
  visible={true}
>
  <styled.div
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bUhCHZ",
            "rules": Array [
              "
	height: 8rem;
	width: 4rem;
	cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: linear-gradient(180deg, rgba(0,0,0,0) 0%, rgba(255,255,255,0.1) 100%);
  border-radius: 3rem;

  @media (min-width: 900px) {
    &:hover {
      background: linear-gradient(180deg, rgba(0,0,0,0) 0%, rgba(255,255,255,0.3) 100%);
    }
  }
  
  @media (max-width: 600px) {
    height: 90%;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <div
        className="sc-bdVaJa bUhCHZ"
        onClick={[Function]}
      >
        <styled.div
          visible={true}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "cLAQHZ",
                  "rules": Array [
                    "
  height: 1.4rem;
  width: 1rem;
  border-radius: 40%;
  background: rgb(89,21,12);
  background: linear-gradient(126deg, rgba(89,21,12,1) 0%, rgba(238,141,101,1) 0%, rgba(193,104,66,1) 37%, rgba(213,49,28,1) 39%, rgba(71,19,12,1) 100%);
  opacity: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            visible={true}
          >
            <div
              className="sc-htpNat cLAQHZ"
            />
          </StyledComponent>
        </styled.div>
        <styled.div
          visible={true}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "iUthcb",
                  "rules": Array [
                    "
	height: 5rem;
	width: 0.6rem;
	background: rgb(224,175,76);
  background: linear-gradient(270deg, rgba(224,175,76,1) 0%, rgba(224,175,76,1) 50%, rgba(230,186,102,1) 50%, rgba(237,201,136,1) 100%);
  cursor: pointer;
  opacity: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            visible={true}
          >
            <div
              className="sc-bwzfXH iUthcb"
            />
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</_default>
`;
